src/main/java/net/minestom/server/ping/Status.java

```diff
@@ -0,0 +1,224 @@
+            }
+
+            @Contract(value = "_ -> this")
+            public Builder maxPlayers(int maxPlayers) {
+                this.maxPlayers = maxPlayers;
+                return this;
+            }
+
+            @Contract(value = "_ -> this")
+            public Builder sample(List<@NotNull NamedAndIdentified> sample) {
+                this.sample = sample;
+                return this;
+            }
+
+            @Contract(value = "_ -> this")
+            public Builder sample(NamedAndIdentified profile) {
+                this.sample.add(profile);
+                return this;
+            }
+
+            @Contract(value = "_ -> this")
+            public Builder sample(GameProfile profile) {
+                return this.sample(NamedAndIdentified.of(profile.name(), profile.uuid()));
+            }
+
+            @Contract(value = "_ -> this")
+            public Builder sample(Component component) {
+                return this.sample(NamedAndIdentified.named(component));
+            }
+
+            @Contract(value = "_ -> this")
+            public Builder sample(String string) {
+                return this.sample(NamedAndIdentified.named(string));
+            }
+
+            public @NotNull PlayerInfo build() {
+                return new PlayerInfo(this.onlinePlayers, this.maxPlayers, this.sample);
+            }
+        }
+    }
+
+    public static final class Builder {
+        public static final Component DEFAULT_DESCRIPTION = Component.text("Minestom Server");
+
+        private @NotNull Component description;
+        private byte @Nullable [] favicon;
+        private @NotNull VersionInfo versionInfo;
+        private @Nullable PlayerInfo playerInfo;
+        private boolean enforcesSecureChat;
+
+        private Builder() {
+            this.description = DEFAULT_DESCRIPTION;
+            this.versionInfo = VersionInfo.DEFAULT;
+            this.playerInfo = PlayerInfo.online();
+        }
+
+        private Builder(Status status) {
+            this.description = status.description;
+            this.favicon = status.favicon;
+            this.versionInfo = status.versionInfo;
+            this.playerInfo = status.playerInfo;
+            this.enforcesSecureChat = status.enforcesSecureChat;
+        }
+
+        @Contract(value = "_ -> this")
+        public Builder description(@NotNull Component description) {
+            this.description = description;
+            return this;
+        }
+
+        @Contract(value = "_ -> this")
+        public @NotNull Builder favicon(byte @Nullable [] favicon) {
+            this.favicon = favicon;
+            return this;
+        }
+
+        @Contract(value = "_ -> this")
+        public @NotNull Builder versionInfo(@NotNull VersionInfo versionInfo) {
+            this.versionInfo = versionInfo;
+            return this;
+        }
+
+        @Contract(value = "_ -> this")
+        public @NotNull Builder playerInfo(@Nullable PlayerInfo playerInfo) {
```

Comment:
Should add an alias for `playerInfo` which takes just player counts as thats the most common case